### PR TITLE
[SPARK-43243][PYTHON][CONNECT] Add level param to printSchema for Python

### PR DESCRIPTION
**Pull request by khalidmammadov**
### What changes were proposed in this pull request?
This feature parity improvement and to add **level** param to df.printSchema for Python API (PySpark & Connect)

## Connect API

Examples:
```
root@f53642e9adb0:/home/spark# bin/pyspark --remote "local[*]"
Python 3.9.5 (default, Nov 23 2021, 15:27:38) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
23/04/23 13:31:58 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /__ / .__/\_,_/_/ /_/\_\   version 3.5.0.dev0
      /_/

Using Python version 3.9.5 (default, Nov 23 2021 15:27:38)
Client connected to the Spark Connect server at localhost
SparkSession available as 'spark'.
>>> df = spark.createDataFrame([(1, (2,2))], ["a", "b"])
>>> df.printSchema(1)
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)

>>> df.printSchema(2)
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)
 |    |-- _1: long (nullable = true)
 |    |-- _2: long (nullable = true)

>>> df.printSchema(3)
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)
 |    |-- _1: long (nullable = true)
 |    |-- _2: long (nullable = true)
 
>>> df.printSchema()
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)
 |    |-- _1: long (nullable = true)
 |    |-- _2: long (nullable = true)
```
## PySpark

```
root@f53642e9adb0:/home/spark# bin/pyspark                    
Python 3.9.5 (default, Nov 23 2021, 15:27:38) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
23/04/23 13:36:42 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /__ / .__/\_,_/_/ /_/\_\   version 3.5.0-SNAPSHOT
      /_/

Using Python version 3.9.5 (default, Nov 23 2021 15:27:38)
Spark context Web UI available at http://localhost:4040
Spark context available as 'sc' (master = local[*], app id = local-1682257002957).
SparkSession available as 'spark'.
>>> df = spark.createDataFrame([(1, (2,2))], ["a", "b"])
>>> df.printSchema(1)
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)

>>> df.printSchema(2)
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)
 |    |-- _1: long (nullable = true)
 |    |-- _2: long (nullable = true)

>>> df.printSchema(3)
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)
 |    |-- _1: long (nullable = true)
 |    |-- _2: long (nullable = true)

>>> df.printSchema(0)
root
 |-- a: long (nullable = true)
 |-- b: struct (nullable = true)
 |    |-- _1: long (nullable = true)
 |    |-- _2: long (nullable = true)

```

### Why are the changes needed?
Feature parity

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
Existing and new test cases

### CODE REVIEW

**Review by zhengruifeng — APPROVED** [review]
LGTM, also cc @HyukjinKwon

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -575,14 +575,23 @@ def schema(self) -> StructType:
...
         .. versionchanged:: 3.4.0
             Supports Spark Connect.
 
+        .. versionchanged:: 3.5.0
+            Added Level parameter.
+
+        Parameters
+        ----------
+        level : int, optional, default None
+            How many levels to print for nested schemas.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/dataframe.py:593]: Can we move:

```
        .. versionchanged:: 3.5.0
            Added Level parameter.
```

to here?

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/dataframe.py [hunk 2]
```diff
@@ -584,14 +584,14 @@ def printSchema(self, level: Optional[int] = None) -> None:
...
 
-        .. versionchanged:: 3.5.0
-            Added Level parameter.
-
         Parameters
         ----------
         level : int, optional, default None
             How many levels to print for nested schemas.
 
+        .. versionchanged:: 3.5.0
```
Comment (zhengruifeng) [inline python/pyspark/sql/dataframe.py:592]: I think the indent is not right, you may refer to https://github.com/apache/spark/blob/7f724c3bc7567b0cddc09d5bed11b79879533368/python/pyspark/sql/dataframe.py#L1917-L1921

Comment (khalidmammadov) [reply python/pyspark/sql/dataframe.py]: Good spot! fixed. Thanks

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**